cmd/connqc/client.go

```diff
@@ -19,10 +22,19 @@ func runClient(c *cli.Context) error {
 	readTimeout := c.Duration(flagReadTimeout)
 	writeTimeout := c.Duration(flagWriteTimeout)
 
-	sig := connqc.NewClient(backoff, sendInterval, readTimeout, writeTimeout, log)
+	var protocol string
+	switch {
+	case c.Bool(flagProtocolTCP) && !c.Bool(flagProtocolUDP):
+		protocol = "tcp"
+	case c.Bool(flagProtocolUDP) && !c.Bool(flagProtocolTCP):
+		protocol = "udp"
+	default:
+		return fmt.Errorf("either --%s or --%s must be set", flagProtocolTCP, flagProtocolUDP)
+	}
+	log = log.With(lctx.Str("protocol", protocol))
```

Comment:
Wouldn't it be simpler to have one `--protocol` flag that would be required and allow us to choose between the two protocols? Do we ever want to start both at the same time? If we do, it would be clearer to do it using two `connqc` processes, WDYT?

Either that or make the protocol a subcommand instead of a flag, i.e. `./server udp --addr=<xxx>` / `./server tcp --addr=<xxx>`.

Reply:
There was a discussion around this in Slack (without your input, so you might have missed that?), where we agreed on this approach, at least from my understanding.

Now that I think about it, I mainly voted for `--udp` & `--tcp` to allow both connections at once, which is _not_ supported in this implementation haha. As @DrPsychick also insisted that `--tcp` & `--udp` "makes it more complicated", I will update this to `--protocol` with a default to `tcp`.

Reply:
I think we concluded that there are use cases where one would want to send/receive UDP and TCP on the same port, but I would do this in a separate PR now. If one needs to test both, one can run separate instances on 2 different ports.

Reply:
The server accepts TCP and UDP on the same port, these params were just for the client.

Reply:
As you prefer, I just thought this could simplify things but if there is a justification for it I am fine with it 👍

In the future make sure to talk about those things on the issue rather than on Slack though, to make the repository more transparent/open 😁